### PR TITLE
:seedling: Gitlab search API tests

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Add tests for `search()` function, including valid response and error handling

#### What kind of change does this PR introduce?

(Is it a bug fix, feature, docs update, something else?)

- [ ] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note

```

### CODE REVIEW

File: clients/gitlabrepo/search_test.go [hunk 1]
```diff
@@ -125,3 +129,69 @@ func TestBuildQuery(t *testing.T) {
...
+
+			repoURL := repoURL{
+				owner:     "ossf-tests",
+				commitSHA: clients.HeadSHA,
+			}
+			handler.init(&repoURL)
+			got, err := handler.search(clients.SearchRequest{
+				Query:    "testquery",
+				Filename: "filename1.txt",
+				Path:     "dir1/dir2",
```
Comment (spencerschrock) [inline clients/gitlabrepo/search_test.go:186]: Do you think it matters that the query doesn't line up with the README.md file being requested?

File: clients/gitlabrepo/search_test.go [hunk 2]
```diff
@@ -125,3 +129,69 @@ func TestBuildQuery(t *testing.T) {
...
+				Results: []clients.SearchResult{
+					{
+						Path: "README.md",
+					},
+				},
+				Hits: 1,
+			},
+
+			wantErr: false,
+		},
```
Comment (spencerschrock) [inline clients/gitlabrepo/search_test.go:154]: can you add a test which returns 0 results?

**Review by spencerschrock — APPROVED** [review]

File: clients/gitlabrepo/search_test.go (hunk 1 above)
Comment (naveensrinivasan) [reply clients/gitlabrepo/search_test.go:186]: I don't think so.

File: clients/gitlabrepo/search_test.go (hunk 2 above)
Comment (naveensrinivasan) [reply clients/gitlabrepo/search_test.go:154]: 👍

**Pull request merged**